### PR TITLE
Adding ECDSA support

### PR DESCRIPTION
**Pull request by diogomonica**
Signed-off-by: Diogo Monica diogo@docker.com

### CODE REVIEW

**Comment by aaronlehmann** [comment]
I'd suggest factoring out the common parts of the two verifiers. But otherwise, LGTM. I like the logging you included. This is basically the same as the verifiers I wrote, but the logging makes it better.

File: signed/verifiers.go [hunk 1]
```diff
@@ -137,3 +141,86 @@ func (v RSAPSSX509Verifier) Verify(key data.Key, sig []byte, msg []byte) error {
...
+		return ErrInvalid
+	}
+
+	ecdsaPubKey, ok := cert.PublicKey.(*ecdsa.PublicKey)
+	if !ok {
+		logrus.Infof("Value was not an RSA public key")
+		return ErrInvalid
+	}
+
+	sigLength := len(sig)
```
Comment (endophage) [inline signed/verifiers.go]: This line to the end of the function appears to be replicated in the ECDSAVerifier. Would be good to refactor it out but not critical (if it works, it's never going to change)

**Comment by endophage** [comment]
Tests, otherwise LGTM.

**Comment by endophage** [comment]
Looks good. Added an issue to create tests for the verifiers in general.

**Pull request merged**